src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -665,6 +755,27 @@ private List<SyntaxNode> VerifyAndPermuteParamNodes(IEnumerable<XmlElementSyntax
         }
 
         protected override IEnumerable<AbstractFormattingRule> GetFormattingRules(Document document)
-            => SpecializedCollections.SingletonEnumerable(new ChangeSignatureFormattingRule()).Concat(Formatter.GetDefaultFormattingRules(document));
+            => Formatter.GetDefaultFormattingRules(document).Concat(new ChangeSignatureFormattingRule());
+
+        protected override SyntaxNode AddNameToArgument(SyntaxNode newArgument, string name)
+        {
+            return (newArgument as ArgumentSyntax).WithNameColon(NameColon(name));
```

Comment:
```suggestion
            return ((ArgumentSyntax)newArgument).WithNameColon(NameColon(name));
```

Reply:
it woudl also be desirable to use generics here to make this strongly typed so we don't even need this.

Reply:
how does this work with attribute arguments?  won't this crash with an null-ref exception?

Reply:
Fixed the attribute arguments part.